daemon/prompter.c

```diff
@@ -741,6 +741,7 @@ prompter_invocation_args(const prompter_t *self, appinfo_t *appinfo)
     g_variant_builder_add(&builder, "{s^as}", "required", vector);
     g_strfreev(vector);
 
+    // TODO: This may need some adjustment..!?
```

Comment:
Then again, if /etc/sailjail/applications doesn't contain any desktop files for anything that would be prompted (launched from launcher or other UI buttons), it might be that this works as is just okay. Comments are welcome.

Reply:
Actually, it does accept those as well but IMO it probably shouldn't. I think this check that you described would make a lot of sense so I'll just add one there. :+1:

Reply:
How about adding another value to appinfo_t? It something like anf_promptable that would be true iff the file exists in the standard directory?

Reply:
Added the check as it seemed to me to make the most sense. Having that extra member in `appinfo_t` might have some other benefits but it probably shouldn't be called `promptable`.

Regarding the implementation: While I put the check inside `prompter_invocation_args()` as that's where the desktop file path is needed, errors are logged only after returning so in that case the path needs constructed twice anyway but at least it's nicely inside if block. I also considered passing desktop path to `prompter_invocation_args()` but since that can be deduced from `appinfo_t` I didn't see it as a useful thing. Also I didn't notice any need for managing `GVariant` reference counts in any special way. Either they are constructed as before or they aren't and the pointer is `NULL`. Did I miss something?

Reply:
> Did I miss something?

Probably not. It is just that *I* am not yet comfortable with GVariant refcounting details and got lazy ;-)

But continuing the thought: If we know such prompting will fail (or we choose not to do it) -> it is already useless to feed it into prompter queue -> such desktop file exists check could be done already in service_dbus_call_cb(). [And I'd consider having it in both as a plus: early rejection and re-evaluation in case the situation changes while the request is sitting in the queue].

Reply:
Added the check to service side too.